### ISSUE TITLE
Cross compiling without QWidget fails in v2.5.1

### ISSUE THREAD

**Issue by JackDesBwa** [KDAB/GammaRay#248]
I followed [Cross compiling GammaRay](https://github.com/KDAB/GammaRay/wiki/Cross-compiling-GammaRay) wiki page to build gammaray on my embedded system, especially the buildroot section since this is the tool I use.

The page proposes to modify gammaray.mk ; here are my changes:
GAMMARAY_VERSION = v2.5.1
GAMMARAY_DEPENDENCIES = qt5base qt5multimedia qt5serialport qt5declarative qt5quickcontrols

Gammaray did not compile ; unfortunately, I did not note the exact error but I made it work again.

Seems that he lack of QWidget made style injector to fail to compile.
Thus I made these changes to disable style injector when there is no QWidget available:

```
diff --git a/launcher/injector/styleinjector.cpp b/launcher/injector/styleinjector.cpp
index f35059e..731e8f6 100644
--- a/launcher/injector/styleinjector.cpp
+++ b/launcher/injector/styleinjector.cpp
@@ -76,7 +76,7 @@ bool StyleInjector::launch(const QStringList &programAndArgs, const QString &pro
 
 bool StyleInjector::selfTest()
 {
-#ifndef GAMMARAY_CORE_ONLY_LAUNCHER
+#if !defined(GAMMARAY_CORE_ONLY_LAUNCHER) && defined(HAVE_QT_WIDGETS)
     QCoreApplication::addLibraryPath(Paths::currentPluginsPath());
     if (!QStyleFactory::keys().contains(QStringLiteral("gammaray-injector"))) {
         mErrorString = tr("Injector style plugin is not found in the Qt style "
diff --git a/launcher/main.cpp b/launcher/main.cpp
index b4caeff..5dbca0c 100644
--- a/launcher/main.cpp
+++ b/launcher/main.cpp
@@ -186,7 +186,7 @@ int main(int argc, char **argv)
     for (int i = 1; i < argc; ++i)
         args.push_back(QString::fromLocal8Bit(argv[i]));
 
-#ifndef GAMMARAY_CORE_ONLY_LAUNCHER
+#if !defined(GAMMARAY_CORE_ONLY_LAUNCHER) && defined(HAVE_QT_WIDGETS)
     QApplication app(argc, argv); // for style inspector
 #else
     QCoreApplication app(argc, argv);
```

Not sure it is the right thing to do, but it allowed me to inspect my application on target through network.

**Comment by vkrause** [comment]
The build issue has been fixed in e4ccc0017031c7bb9a9075b5a25d20a974c5c6f4 already.

**Comment by JackDesBwa** [comment]
Didn't think about testing with dev version, sorry.
Thanks for reference.
